site/src/xServices/terminal/terminalXService.ts

```diff
@@ -170,9 +171,11 @@ export const terminalMachine =
               return reject("workspace agent is not set")
             }
             const proto = location.protocol === "https:" ? "wss:" : "ws:"
-            const socket = new WebSocket(
-              `${proto}//${location.host}/api/v2/workspaceagents/${context.workspaceAgent.id}/pty?reconnect=${context.reconnection}`,
-            )
+            let url = `${proto}//${location.host}/api/v2/workspaceagents/${context.workspaceAgent.id}/pty?reconnect=${context.reconnection}`
```

Comment:
We try to avoid `let`. How about making a const that's either the command query or '', and interpolating it in `url`?

Reply:
Why do we avoid using `let`?

Reply:
Because it allows mutation.

Reply:
I understand, but only inside the function where the variable is defined. Why is the mutation considered bad? In this case, I'd argue it's cleaner code (unless I'm misunderstanding what you're proposing, which is entirely possible ;p).

Reply:
I like functional programming, so I try to avoid mutation whenever possible. I find it's often at the root of annoying bugs. Here's a function from v1 where I think reassignment (with `let` but also of the argument) made it hard to read and refactor https://github.com/coder/v1/blob/master/product/coder/site/src/components/Workspaces/CreateEditWorkspaceForm/hooks.ts#L102

It's not a big deal here, but for clarity my proposal is
```
const commandQuery = context.command ? `&command=${encodeURIComponent(context.command)}` : ""
const url = `${proto}//${location.host}/api/v2/workspaceagents/${context.workspaceAgent.id}/pty?reconnect=${context.reconnection}${commandQuery}`
```

But it's probably better to do it this way anyway https://www.valentinog.com/blog/url/#how-to-build-an-url-and-its-search-parameters

Reply:
Fixed!